firefox-ios/Client/Application/WindowManager.swift

```diff
@@ -30,29 +30,34 @@ protocol WindowManager {
 
     /// Signals the WindowManager that a window was closed.
     /// - Parameter uuid: the ID of the window.
-    func windowDidClose(uuid: WindowUUID)
+    func windowWillClose(uuid: WindowUUID)
 
     /// Supplies the UUID for the next window the iOS app should open. This
     /// corresponds with the window tab data saved to disk, or, if no data is
-    /// available it provides a new UUID for the window.
+    /// available it provides a new UUID for the window. The resulting UUID
+    /// is then "reserved" in order to ensure that during app launch if multiple
+    /// windows are being restored concurrently, we never supply the same UUID
+    /// to more than one window.
     /// - Returns: a UUID for the next window to be opened.
-    func nextAvailableWindowUUID() -> WindowUUID
-}
-
-/// Abstract protocol that any Coordinator can conform to in order to respond
-/// to key window lifecycle events, such as cleaning up when a window is closed.
-protocol WindowEventCoordinator {
-    /// Notifies the coordinator that its parent window/scene is being removed.
-    func coordinatorWindowWillClose()
+    func reserveNextAvailableWindowUUID() -> WindowUUID
+
+    /// Signals the WindowManager that a window event has occurred. Window events
+    /// are communicated to any interested Coordinators for _all_ windows, but
+    /// any one event is always associated with one window in specific. This window
+    /// can be identified by the `windowUUID` parameter.
+    /// - Parameter event: the event that occurred and any associated metadata.
+    func postWindowEvent(event: WindowEvent, windowUUID: WindowUUID)
```

Comment:
nit; `windowUUID` parameter should be in the doc string as well

Reply:
Updated and added the `windowUUID`.